tests/integration/test_execution.py

```diff
@@ -885,3 +876,73 @@ def test_should_publish_messages_if_connection_drops_when_not_transactions():
     # the two message should have been published due to retry (but no transactions involved)
     assert queue_status.messages_enqueued == 2
     assert queue_status.number_of_pending_messages == 2
+
+
+@pytest.mark.django_db
+def test_should_open_a_new_db_connection_when_previous_connection_is_obsolete_or_unusable(settings):
+    # Arrange - listen for db connection created signal with mocked function
+    new_db_connection_callback = mock.MagicMock(return_value=None)
+    connection_created.connect(new_db_connection_callback)
+
+    # Arrange - settings CONN_MAX_AGE to zero, forcing connection renewal for every new message
+    settings.DATABASES["default"]["CONN_MAX_AGE"] = 0
+
+    # Arrange - random destination and fake arbitrary message body
+    destination = f"/queue/new-connections-{uuid4()}"
+    body = {"key": "value"}
+    arbitrary_number_of_msgs = 5
+
+    # Act - start listening random destination queue and publish arbitrary number of messages to it
+    listener = start_processing(
+        destination, callback_with_explicit_db_connection_path, is_testing=True, return_listener=True
+    )
+
+    with build_publisher(f"random-publisher-{uuid4()}").auto_open_close_connection() as publisher:
+        for _ in range(arbitrary_number_of_msgs):
+            publisher.send(body, destination)
+
+    sleep(0.5)  # some sleep to give enough time to process the messages
+
+    # Assert - for every new message a new db connection is created
+    assert new_db_connection_callback.call_count == arbitrary_number_of_msgs
+
+    # Assert - all threads that estabilished db connection, reseted it
+    threads_with_db_connections = filter(lambda t: hasattr(t, "db"), threading.enumerate())
+    assert all(t.db.connection is None for t in threads_with_db_connections)
```

Comment:
```suggestion
    threads_with_db_connections = filter(lambda t: hasattr(t, "db"), threading.enumerate())
    threads_with_db_connections = list(threads_with_db_connections)
    assert len(threads_with_db_connections) == 1
    assert all(t.db.connection is None for t in threads_with_db_connections)
```

**suggestion (non-blocking):** If the list is empty, the test will pass anyway, then is it a good idea to assert it has one item at least?

Reply:
Yep! It certainly is! With that, we can guarantee that our defined callback is really setting a new attribute (`db`) and/or that the thread is still alive.

As we want an iterable that isn't an iterator (since `any` and `all` relies on it), what do you think of using list-comprehension instead?

```suggestion
    threads_with_db_connections = [t for t in threading.enumerate if hasattr(t, "db")]
    assert any(threads_with_db_connections)
    assert all(t.db.connection is None for t in threads_with_db_connections)
```

Reply:
I couldn't agree more!